### PR TITLE
Feature/e2e subscribe unsubscribe from channel

### PR DESCRIPTION
**Pull request by vlad-dargel**

### CODE REVIEW

File: tests/e2e/pageObjects/pub-sub-page.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    publishButton = Selector('[data-testid=publish-message-submit]');
+    clearPubSubButton = Selector('[data-testid=clear-pubsub-btn]');
+    scrollDownButton = Selector('[data-testid=messages-list-anchor-btn]');
+    //INPUTS
+    channelNameInput = Selector('[data-testid=field-channel-name]');
+    messageInput = Selector('[data-testid=field-message]');
+
+    /**
+     * Publish message in pubsub
+     * @param chanel The name of chanel
```
[the remark this answers is not in the document]
Comment (vlad-dargel) [reply tests/e2e/pageObjects/pub-sub-page.ts:31]: thanks! fixed

File: tests/e2e/tests/critical-path/pub-sub/debug-mode.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        //Scroll to the first messages
+        await t.scrollIntoView(pubSubPage.pubSubPageContainer.find(pubSubPage.cssSelectorMessage).withText('first'));
+        await pubSubPage.publishMessage('test', 'message to scroll');
+        await t.scrollIntoView(pubSubPage.pubSubPageContainer.find(pubSubPage.cssSelectorMessage).withText('message to scroll'));
+        await cliPage.sendCommandInCli('20 publish chanel fourth');
+        //Verify auto-scroll resumes automatically
+        await t.expect(pubSubPage.pubSubPageContainer.find(pubSubPage.cssSelectorMessage).withText('fourth').visible).ok('Message is not visible');
+    });
+test
+    .meta({ env: env.web, rte: rte.standalone })('Verify that user can get to the newest message in one click', async t => {
```
[the remark this answers is not in the document]
Comment (vlad-dargel) [reply tests/e2e/tests/critical-path/pub-sub/debug-mode.e2e.ts]: fixed, thanks!

File: tests/e2e/tests/critical-path/pub-sub/subscribe-unsubscribe.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        await t.expect(pubSubPage.subscribeStatus.textContent).eql('You are not subscribed', 'User is not unsubscribed');
+        await t.expect(pubSubPage.subscribeButton.textContent).eql('Subscribe', 'Subscribe button is not displayed');
+    });
+test('Verify that when user subscribe to the pubsub channel he can see all the messages being published to my database from the moment of my subscription', async t => {
+        //Subscribe to channel
+        await t.click(pubSubPage.subscribeButton);
+        await t.expect(pubSubPage.subscribeStatus.textContent).eql('You are  subscribed', 'User is not subscribed', { timeout: 10000 });
+        //Verify that user can publish a message to a channel
+        await pubSubPage.publishMessage('test', 'published message');
+        await t.expect(pubSubPage.pubSubPageContainer.find(pubSubPage.cssSelectorMessage).withText('published message').exists).ok('Message is not displayed');
```
[the remark this answers is not in the document]
Comment (vlad-dargel) [reply tests/e2e/tests/critical-path/pub-sub/subscribe-unsubscribe.e2e.ts]: fixed, thanks! added verifyMessageDisplaying(message: string, displayed: boolean) function

File: tests/e2e/tests/critical-path/pub-sub/subscribe-unsubscribe.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        await t.expect(pubSubPage.clearButtonTooltip.textContent).contains('Clear Messages', 'Clear Messages tooltip not displayed');
+        await t.click(pubSubPage.clearPubSubButton);
+        //Verify that the clear of the messages does not affect the subscription state
+        await t.expect(pubSubPage.subscribeStatus.textContent).eql('You are  subscribed', 'User is not subscribed', { timeout: 10000 });
+        //Verify that the Messages counter is reset after clear messages
+        await t.expect(pubSubPage.totalMessagesCount.textContent).contains('0', 'Total counter value is incorrect');
+        //Verify messages are cleared
+        await t.expect(pubSubPage.pubSubPageContainer.find(pubSubPage.cssSelectorMessage).withText('message').exists).notOk('Message is still displayed');
+        await t.expect(pubSubPage.pubSubPageContainer.find(pubSubPage.cssSelectorMessage).withText('second m').exists).notOk('Second message is still displayed');
+    });
```
[the remark this answers is not in the document]
Comment (vlad-dargel) [reply tests/e2e/tests/critical-path/pub-sub/subscribe-unsubscribe.e2e.ts]: fixed, thanks! replaced by verifyMessageDisplaying func

**Review by ofersteinberg — APPROVED** [review]

File: tests/e2e/pageObjects/pub-sub-page.ts [hunk 5]
```diff
@@ -49,16 +49,4 @@ export class PubSubPage {
         await this.publishMessage(channel, message);
         await t.expect(this.pubSubPageContainer.find('[data-testid^=row]').withText('message').exists).ok('Message is not displayed');
     }
-
-    /**
-     * Verify message is/not displayed
-     * @param message The text of message
-     * @param displayed True if should be displayed, false if shouldn't
-     */
-    async verifyMessageDisplaying(message: string, displayed: boolean): Promise<void> {
```
Comment (ofersteinberg) [inline tests/e2e/pageObjects/pub-sub-page.ts]: Why was it removed?
I see you used it on various tests.
The only thing is that the location here is probably wrong.

Comment (vlad-dargel) [reply tests/e2e/pageObjects/pub-sub-page.ts]: I moved it directly in tests, to not store it into pageObjects

Comment (ofersteinberg) [reply tests/e2e/pageObjects/pub-sub-page.ts]: But now it's on each test. What will happen when you'll have modified it? Will you do that X times?
Add it somewhere on `tests/e2e/helpers`

Comment (vlad-dargel) [reply tests/e2e/pageObjects/pub-sub-page.ts]: Yes, it makes sense, thanks, fixed

**Review by ofersteinberg — APPROVED** [review]

**Pull request merged**